src/main.cpp

```diff
@@ -6239,6 +6239,17 @@ bool static ProcessMessage(const CChainParams& chainparams, CNode* pfrom, string
             return true;
         }
 
+        // If we already know the last header in the message, then it contains
```

Comment:
This looks okay to me. I think it would be better to move this block of code down below the next block where the chaining of headers is checked, simply because it relies upon the ordering of the headers and its result won't be used until `AcceptBlockHeader` has succeeded anyway. 

Also minor nits on the CLI flag: `-fixibd` is not adequately descriptive, I'd prefer `-no-download-redundant-headers` or something of the sort, and also I'd like to not introduce additional locations where CLI flags are referenced directly because network code shouldn't be directly coupled to CLI arguments; ideally these flags would be passed to this function as an argument. If that's too significant of a change, no big deal.

Reply:
> I think it would be better to move this block of code down below the next block...

[`AcceptBlockHeader()`](https://github.com/zcash/zcash/blob/8c000ae281b5be03dba189ff26c90f95ca8caf05/src/main.cpp#L4321) (called in the block below) (usually) adds the block to the index, so that would break the test we're making here (i.e., `hasNewHeaders` would always be set to false).

> I'd prefer `-no-download-redundant-headers`

I think it's best if we don't start argument names with `-no` because upstream has a [feature](https://github.com/bitcoin/bitcoin/blob/9158d6f34153dc937e109d0e7486e956a26f20aa/src/util/system.cpp#L180) that we haven't pulled in yet that gives a special meaning to such arguments. (It automatically converts `-nofoo=0` to `-foo=1`.)

> I'd like to not introduce additional locations where CLI flags are referenced directly

I think removing the direct access to configuration options could be done in a separate PR so it's done consistently; otherwise, I need to decide how to do it right now. I see there are five existing instances of that in this function (`ProcessMessage()`).